core/mutator/entry/entry_test.go

```diff
@@ -183,13 +182,35 @@ func TestCheckMutation(t *testing.T) {
 		{key, entryData1, entryData2, hashEntry1[:], signers1, nil},                                       // Second mutation, missing current signature, should work
 	} {
 		// Prepare mutations.
-		mutation, err := prepareMutation(tc.key, tc.entryData, tc.previous, tc.signers)
+		mutation, err := prepareMutation(tc.key, tc.newEntry, tc.previous, tc.signers)
 		if err != nil {
-			t.Fatalf("prepareMutation(%v, %v, %v)=%v", tc.key, tc.entryData, tc.previous, err)
+			t.Fatalf("prepareMutation(%v, %v, %v)=%v", tc.key, tc.newEntry, tc.previous, err)
+		}
+
+		if _, got := New().Mutate(tc.oldEntry, mutation); got != tc.err {
+			t.Errorf("Mutate(%v, %v)=%v, want %v", tc.oldEntry, mutation, got, tc.err)
 		}
+	}
+}
 
-		if _, got := New().Mutate(tc.oldValue, mutation); got != tc.err {
-			t.Errorf("Mutate(%v, %v)=%v, want %v", tc.oldValue, mutation, got, tc.err)
+func TestFromLeafValue(t *testing.T) {
+	entry := &tpb.Entry{Commitment: []byte{1, 2}}
+	entryB, _ := proto.Marshal(entry)
+	for i, tc := range []struct {
+		leafVal []byte
+		want    *tpb.Entry
+		wantErr bool
+	}{
+		{make([]byte, 0), nil, false},            // empty leaf -> nil, no error
```

Comment:
Add a test case for both `nil` and `[]byte{}` and make sure they produce different values